src/extra/widgets/animbtn/lv_animbtn.h

```diff
@@ -0,0 +1,118 @@
+/**
+ * @file lv_animbtn.h
+ *
+ */
+
+#ifndef LV_ANIMBTN_H
+#define LV_ANIMBTN_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "../../../lvgl.h"
+
+#if LV_USE_RLOTTIE != 0 && LV_USE_ANIMBTN != 0
+
+/*********************
+ *      DEFINES
+ *********************/
+typedef enum {
+    LV_ANIMBTN_STATE_RELEASED,
+    LV_ANIMBTN_STATE_PRESSED,
+    LV_ANIMBTN_STATE_DISABLED,
+    LV_ANIMBTN_STATE_CHECKED_RELEASED,
+    LV_ANIMBTN_STATE_CHECKED_PRESSED,
+    LV_ANIMBTN_STATE_CHECKED_DISABLED,
+    _LV_ANIMBTN_STATE_NUM,
+} lv_animbtn_state_t;
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+/*State status for anim button*/
+typedef struct {
+    size_t first_frame;
+    size_t last_frame;
+    size_t control; /* A lv_rlottie_ctrl_t instance and only checking Forward/Backward and Loop here */
+} lv_animbtn_state_desc_t;
+
+/*Data of anim button*/
+typedef struct {
+    lv_obj_t obj;
+    lv_animbtn_state_desc_t state_desc[_LV_ANIMBTN_STATE_NUM];
+    lv_obj_t * lottie;
+    lv_animbtn_state_t prev_state;
+} lv_animbtn_t;
+
+extern const lv_obj_class_t lv_animbtn_class;
+
+/**********************
+ * GLOBAL PROTOTYPES
+ **********************/
+
+/**
+ * Create an animated button object
+ * @param parent pointer to an object, it will be the parent of the new animation button
+ * @param anim  pointer to a lv_rlottie object used as backend of this button
+ * @return pointer to the created anim button
+ */
+lv_obj_t * lv_animbtn_create(lv_obj_t * parent, lv_obj_t * anim);
+
+/*======================
+ * Add/remove functions
+ *=====================*/
+
+/*=====================
+ * Setter functions
+ *====================*/
+
+/**
+ * Set animation for a state of the animation button
+ * @param animbtn pointer to an animation button object
+ * @param state for which state set the new animation
+ * @param desc description of what to do when the button is in this state
```

Comment:
Please mention if it can be a local variable that can go out of the scope or not.

Reply:
I don't get you. There's no pointer (except for the animbtn instance, obviously), so the usual copy by value rules apply here.